nucypher/blockchain/eth/agents.py

```diff
@@ -888,19 +888,20 @@ def _contract_name_to_agent_name(name: str) -> str:
         return agent_name
 
     @classmethod
-    def get_agent_by_contract_name(cls,
-                                   contract_name: str,
-                                   registry: BaseContractRegistry,
-                                   eth_provider_uri: Optional[str] = None,
-                                   contract_version: Optional[str] = None
-                                   ) -> EthereumContractAgent:
+    def get_agent_by_contract_name(
+        cls,
+        contract_name: str,
+        registry: BaseContractRegistry,
+        provider_uri: str,
```

Comment:
I suppose there was conversation about renaming it somewhere. Just in case want to say say that initially  we wanted to have different names for staking providers and eth connection providers which lead to not using pure provider name

Reply:
There wasn't really a conversation per se 😅 - `eth_provider_uri` was renamed in some other spots to `provider_uri` since it could be a URI for Ethereum or it could be Polygon, or others in the future. So this made sense to me.

That being said if there is some unintended side effect of this change I'm all ears. Could you elaborate a little on the potential concern/side effect?

Reply:
My point - change `provider_uri` to anything except `provider_uri`, because it's really ambiguous considering staking provider

Reply:
Because in python we have both staking_provider and provider now, and it's not obvious sometimes what exactly meant by provider, especially when you switching between agents

Reply:
Oh I got you, so a naming thing, like rename to `blockchain_provider_uri` or something like that... (cc @KPrasch )